src/Backups/BackupsWorker.cpp

```diff
@@ -178,6 +178,42 @@ namespace
     {
         return status == BackupStatus::RESTORING;
     }
+
+    /// We use slightly different read and write settings for backup/restore
+    /// with a separate throttler and limited usage of filesystem cache.
+    ReadSettings getReadSettingsForBackup(const ContextPtr & context, const BackupSettings & backup_settings)
+    {
+        auto read_settings = context->getReadSettings();
+        read_settings.remote_throttler = context->getBackupsThrottler();
+        read_settings.local_throttler = context->getBackupsThrottler();
+        read_settings.enable_filesystem_cache = false;
+        read_settings.read_from_filesystem_cache_if_exists_otherwise_bypass_cache = backup_settings.read_from_filesystem_cache_if_exists_otherwise_bypass_cache;
```

Comment:
setting `read_from_filesystem_cache_if_exists_otherwise_bypass_cache` will not work if `enable_filesystem_cache` is disabled, see https://github.com/ClickHouse/ClickHouse/blob/5ea0971921ac433df0484927dda39bf4b768f8ca/src/Disks/IO/ReadBufferFromRemoteFSGather.cpp#L21-L23
may be better something like this?
```
read_settings.enable_filesystem_cache = backup_settings.read_from_filesystem_cache_if_exists_otherwise_bypass_cache;
read_settings.read_from_filesystem_cache_if_exists_otherwise_bypass_cache = backup_settings.read_from_filesystem_cache_if_exists_otherwise_bypass_cache;
read_settings.avoid_readthrough_cache_outside_query_context = false; /// not sure if needed
```

Reply:
Thank you!

`avoid_readthrough_cache_outside_query_context` doesn't seem necessary because BACKUP/RESTORE can operate in background but they always have proper query id.